tests/WithMocks.php

```diff
@@ -30,6 +32,22 @@ public function configureMocks()
         $this->faker = app(\Faker\Generator::class);
         $this->faker->addProvider(new \FakerNorthstarId($this->faker));
         $this->faker->addProvider(new \FakerCampaignId($this->faker));
+
+        // Northstar Mock
+        $this->northstarMock = $this->mock(Northstar::class);
+        $this->northstarMock->shouldReceive('asClient')->andReturnSelf();
+        $this->northstarMock->shouldReceive('asUser')->andReturnSelf();
+        $this->northstarMock->shouldReceive('refreshIfExpired')->andReturnSelf();
+        $this->northstarMock->shouldReceive('getUser')->andReturnUsing(function ($type, $id) {
+            return new NorthstarUser([
+                    'id' => $id ? $id : $this->faker->northstar_id,
```

Comment:
Hmm, I think we should make sure this _is_ an ID here before setting it (e.g. if `getUser('email', '...')` was called, we'd still set it as `id` with how this is set up right now). Let's instead check if `$type == 'id'`.

```php
'id' => $type == 'id' ? $id : $this->faker->northstar_id,
```

Reply:
ah that is super smart, thanks for flagging @DFurnes ! Updated that :)